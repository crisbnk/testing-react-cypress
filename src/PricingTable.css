.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 10px;
  padding: 20px;
  box-shadow: 0px 0px 5px 3px rgba(230,230,230,1);
  border-radius: 3%;
}

.title {
  width: 100%;
  margin: 0;
  text-align: left;
}

.price-container {
  width: 100%;
  text-align: left;
}

.price {
  font-size: 25px;
  font-weight: bold;
}

.plan-period {
  font-size: 12px;
  margin-left: 5px;
}

.options-list {
  width: 100%;
  list-style-position: inside;
  margin: 0;
  padding: 30px 0;
  border-top: 1px solid #E8E8E8;
  border-bottom: 1px solid #E8E8E8;
}

.option-item {
  font-size: 12px;
  color: #A5A5A5;
  margin-top: 5px;
  margin-bottom: 5px;
}

.option-item.active {
  color: #000;
}

.button {
  color: #FFF;
  width: 100%;
  height: 30px;
  margin-top: 10px;
  margin-bottom: 5px;
  border: none;
  border-radius: 15px;
  text-transform: uppercase;
}
