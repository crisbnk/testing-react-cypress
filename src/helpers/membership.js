/*
  This is a file with membership page text
*/

exports.membership = {
  "plans": [
    {
      "id": 1,
      "title": "Basic",
      "price": "4,99",
      "buttonColor": "#FF6F74",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": false
        },
        {
          "name": "Free Wifi",
          "active": false
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": false
        },
        {
          "name": "Book classes 8 days in advance",
          "active": false
        },
        {
          "name": "Free friend access",
          "active": false
        }
      ]
    },
    {
      "id": 2,
      "title": "Silver",
      "price": "6,99",
      "buttonColor": "#00D2EC",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": true
        },
        {
          "name": "Free Wifi",
          "active": true
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": false
        },
        {
          "name": "Book classes 8 days in advance",
          "active": false
        },
        {
          "name": "Free friend access",
          "active": false
        }
      ]
    },
    {
      "id": 3,
      "title": "Gold",
      "price": "8,99",
      "buttonColor": "#4172FE",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": true
        },
        {
          "name": "Free Wifi",
          "active": true
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": true
        },
        {
          "name": "Book classes 8 days in advance",
          "active": false
        },
        {
          "name": "Free friend access",
          "active": false
        }
      ]
    },
    {
      "id": 4,
      "title": "Platinum",
      "price": "10,99",
      "buttonColor": "#00FF3B",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": true
        },
        {
          "name": "Free Wifi",
          "active": true
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": true
        },
        {
          "name": "Book classes 8 days in advance",
          "active": true
        },
        {
          "name": "Free friend access",
          "active": true
        }
      ]
    }
  ]
}
