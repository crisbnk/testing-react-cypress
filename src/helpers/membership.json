{
  "plans": [
    {
      "id": 1,
      "title": "Basic",
      "price": "4,99",
      "buttonColor": "rgb(255, 111, 116)",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": false
        },
        {
          "name": "Free Wifi",
          "active": false
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": false
        },
        {
          "name": "Book classes 8 days in advance",
          "active": false
        },
        {
          "name": "Free friend access",
          "active": false
        }
      ]
    },
    {
      "id": 2,
      "title": "Silver",
      "price": "6,99",
      "buttonColor": "rgb(0, 210, 236)",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": true
        },
        {
          "name": "Free Wifi",
          "active": true
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": false
        },
        {
          "name": "Book classes 8 days in advance",
          "active": false
        },
        {
          "name": "Free friend access",
          "active": false
        }
      ]
    },
    {
      "id": 3,
      "title": "Gold",
      "price": "8,99",
      "buttonColor": "rgb(65, 114, 254)",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": true
        },
        {
          "name": "Free Wifi",
          "active": true
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": true
        },
        {
          "name": "Book classes 8 days in advance",
          "active": false
        },
        {
          "name": "Free friend access",
          "active": false
        }
      ]
    },
    {
      "id": 4,
      "title": "Platinum",
      "price": "10,99",
      "buttonColor": "rgb(0, 255, 59)",
      "options": [
        {
          "name": "Unlimited Access to Wellness Club",
          "active": true
        },
        {
          "name": "Free Fitness Training",
          "active": true
        },
        {
          "name": "Free Wifi",
          "active": true
        },
        {
          "name": "Unlimited Use of Hydromassage",
          "active": true
        },
        {
          "name": "Book classes 8 days in advance",
          "active": true
        },
        {
          "name": "Free friend access",
          "active": true
        }
      ]
    }
  ]
}
