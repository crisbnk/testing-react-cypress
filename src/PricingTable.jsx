import React, { Component } from "react";
import "./PricingTable.css";

const PricingTable = props => {
  return (
    <div className={`card package-${props.title.toLowerCase()}`}>
      <h3 className="title">Wellness {props.title}</h3>
      <p className="price-container">
        <span className="price">â‚¬ {props.price}</span>
        <span className="plan-period">/month</span>
      </p>
      <ul className="options-list">
        {props.options.map((opt, index) => {
          return (
            <li
              key={index}
              className={`option-item ${opt.active ? "active" : ""}`}
            >
              {opt.name}
            </li>
          );
        })}
      </ul>
      {props.children}
    </div>
  );
};

export default PricingTable;
