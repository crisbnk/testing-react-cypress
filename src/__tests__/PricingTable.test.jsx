import React from 'react';
import { create } from 'react-test-renderer';
import PricingTable from '../PricingTable';
import { membership } from '../helpers/membership';

describe('Pricing table component', () => {
  test('it matches the snapshot of PricingTable #1', () => {
    const pricingTable1 = membership.plans[0];
    const component = create(
      <PricingTable
        title={pricingTable1.title}
        price={pricingTable1.price}
        options={pricingTable1.options}
        buttonColor={pricingTable1.buttonColor}
      />
    );
    const json = component.toJSON();
    console.log(json);
    expect(json).toMatchSnapshot();
  });
  test('it matches the snapshot of PricingTable #2', () => {
    const pricingTable2 = membership.plans[1];
    const component = create(
      <PricingTable
        title={pricingTable2.title}
        price={pricingTable2.price}
        options={pricingTable2.options}
        buttonColor={pricingTable2.buttonColor}
      />
    );
    const json = component.toJSON();
    console.log(json);
    expect(json).toMatchSnapshot();
  });
  test('it matches the snapshot of PricingTable #3', () => {
    const pricingTable3 = membership.plans[2];
    const component = create(
      <PricingTable
        title={pricingTable3.title}
        price={pricingTable3.price}
        options={pricingTable3.options}
        buttonColor={pricingTable3.buttonColor}
      />
    );
    const json = component.toJSON();
    console.log(json);
    expect(json).toMatchSnapshot();
  });
  test('it matches the snapshot of PricingTable #4', () => {
    const pricingTable4 = membership.plans[3];
    const component = create(
      <PricingTable
        title={pricingTable4.title}
        price={pricingTable4.price}
        options={pricingTable4.options}
        buttonColor={pricingTable4.buttonColor}
      />
    );
    const json = component.toJSON();
    console.log(json);
    expect(json).toMatchSnapshot();
  });
});
