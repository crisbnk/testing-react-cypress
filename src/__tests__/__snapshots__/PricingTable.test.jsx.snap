// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing table component it matches the snapshot of PricingTable #1 1`] = `
<div
  className="card package-basic"
>
  <h3
    className="title"
  >
    Wellness 
    Basic
  </h3>
  <p
    className="price-container"
  >
    <span
      className="price"
    >
      € 
      4,99
    </span>
    <span
      className="plan-period"
    >
      /month
    </span>
  </p>
  <ul
    className="options-list"
  >
    <li
      className="option-item active"
    >
      Unlimited Access to Wellness Club
    </li>
    <li
      className="option-item "
    >
      Free Fitness Training
    </li>
    <li
      className="option-item "
    >
      Free Wifi
    </li>
    <li
      className="option-item "
    >
      Unlimited Use of Hydromassage
    </li>
    <li
      className="option-item "
    >
      Book classes 8 days in advance
    </li>
    <li
      className="option-item "
    >
      Free friend access
    </li>
  </ul>
</div>
`;

exports[`Pricing table component it matches the snapshot of PricingTable #2 1`] = `
<div
  className="card package-silver"
>
  <h3
    className="title"
  >
    Wellness 
    Silver
  </h3>
  <p
    className="price-container"
  >
    <span
      className="price"
    >
      € 
      6,99
    </span>
    <span
      className="plan-period"
    >
      /month
    </span>
  </p>
  <ul
    className="options-list"
  >
    <li
      className="option-item active"
    >
      Unlimited Access to Wellness Club
    </li>
    <li
      className="option-item active"
    >
      Free Fitness Training
    </li>
    <li
      className="option-item active"
    >
      Free Wifi
    </li>
    <li
      className="option-item "
    >
      Unlimited Use of Hydromassage
    </li>
    <li
      className="option-item "
    >
      Book classes 8 days in advance
    </li>
    <li
      className="option-item "
    >
      Free friend access
    </li>
  </ul>
</div>
`;

exports[`Pricing table component it matches the snapshot of PricingTable #3 1`] = `
<div
  className="card package-gold"
>
  <h3
    className="title"
  >
    Wellness 
    Gold
  </h3>
  <p
    className="price-container"
  >
    <span
      className="price"
    >
      € 
      8,99
    </span>
    <span
      className="plan-period"
    >
      /month
    </span>
  </p>
  <ul
    className="options-list"
  >
    <li
      className="option-item active"
    >
      Unlimited Access to Wellness Club
    </li>
    <li
      className="option-item active"
    >
      Free Fitness Training
    </li>
    <li
      className="option-item active"
    >
      Free Wifi
    </li>
    <li
      className="option-item active"
    >
      Unlimited Use of Hydromassage
    </li>
    <li
      className="option-item "
    >
      Book classes 8 days in advance
    </li>
    <li
      className="option-item "
    >
      Free friend access
    </li>
  </ul>
</div>
`;

exports[`Pricing table component it matches the snapshot of PricingTable #4 1`] = `
<div
  className="card package-platinum"
>
  <h3
    className="title"
  >
    Wellness 
    Platinum
  </h3>
  <p
    className="price-container"
  >
    <span
      className="price"
    >
      € 
      10,99
    </span>
    <span
      className="plan-period"
    >
      /month
    </span>
  </p>
  <ul
    className="options-list"
  >
    <li
      className="option-item active"
    >
      Unlimited Access to Wellness Club
    </li>
    <li
      className="option-item active"
    >
      Free Fitness Training
    </li>
    <li
      className="option-item active"
    >
      Free Wifi
    </li>
    <li
      className="option-item active"
    >
      Unlimited Use of Hydromassage
    </li>
    <li
      className="option-item active"
    >
      Book classes 8 days in advance
    </li>
    <li
      className="option-item active"
    >
      Free friend access
    </li>
  </ul>
</div>
`;
